admin/src/main/scala/za/co/absa/spline/admin/DateTimeUtils.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.admin
+
+import java.time.format.{DateTimeFormatter, DateTimeFormatterBuilder}
+import java.time.temporal.ChronoField
+import java.time.{ZoneId, ZonedDateTime}
+import scala.util.{Failure, Success, Try}
+import scala.util.control.NonFatal
+
+object DateTimeUtils {
+
+  private val ZonedDateTimeRegexp = (s"" +
+    "^" +
+    """([\dT:.+\-]+?)""".r + // local datetime
+    """(Z|[+\-]\d\d:\d\d)?""".r + // timezone offset
+    """(?:\[([\w/+\-]+)])?""".r + // timezone name
+    "$").r
+
+  private val ZonedDateTimeFormatter = new DateTimeFormatterBuilder()
+    .parseCaseInsensitive()
+    .append(DateTimeFormatter.ISO_LOCAL_DATE)
+    .optionalStart()
+    .appendLiteral('T')
+    .append(DateTimeFormatter.ISO_LOCAL_TIME)
+    .optionalEnd()
+    .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+    .toFormatter();
+
+  def parseZonedDateTime(s: String, defaultZoneId: ZoneId = ZoneId.systemDefault): ZonedDateTime =
+    Try {
+      val ZonedDateTimeRegexp(ldt, tzOffset, tzId) = s
+      val maybeTzIds = Seq(tzId, tzOffset).map(Option.apply)
+
+      require(!maybeTzIds.forall(_.isDefined), "Either timezone ID or offset should be specified, not both")
```

Comment:
`ZonedDateTime.parse` allows both offset and name at the same time . I think it's good idea to accept all inputs that are valid for `ZonedDateTime.parse`

Reply:
The issue with allowing only **geographical zone id** is that it is ambiguous. Not unique. 

```
2022-10-30T02:30:00[Europe/Prague] is both:
2022-10-30T02:30+02:00[Europe/Prague]
2022-10-30T02:30+01:00[Europe/Prague]
```

This is the day and hour of change from summer time to winter time:
```scala
val start = ZonedDateTime.parse("2022-10-30T02:30:00+02:00[Europe/Prague]")
val end = start.plusHours(1)
// start: java.time.ZonedDateTime = 2022-10-30T02:30+02:00[Europe/Prague]
// end: java.time.ZonedDateTime =   2022-10-30T02:30+01:00[Europe/Prague]
```

Another interesting behaviour (not so relevant for us though) is this:
```scala
val start = ZonedDateTime.parse("2022-10-30T02:30:00+02:00")
val end = start.plusHours(1)
// start: java.time.ZonedDateTime = 2022-10-30T02:30+02:00
// end: java.time.ZonedDateTime   = 2022-10-30T03:30+02:00
```
When no geographical zone id is provided the ZonedDateTime will stay in the same offset.

Reply:
We agreed to solve this as part of other ticket #1139